### ISSUE TITLE
Includes between headers should be relative

### ISSUE THREAD

**Issue by buchanae** [pezmaster31/bamtools#50]
-- TL;DR --
I think we could improve the experience of linking against bamtools by making #includes between headers within bamtools relative (or some other better fix that probably exists).

Mainly, when I install bamtools, I don't want to modify my environment variables or write special build code to find that library.

If you think it's worth a shot, please let me know and I'll take a crack at writing a patch.

-- Longer version --

I've run into the following problem, and it's becoming a headache...

1) I write a library named Foo that links against bamtools.

It's important that I can install bamtools in a central location, but because of the way bamtools installs, I must add explicitly modify my environment variables to include the bamtools include and lib dirs (or I could write some special build code to include the correct paths).

This means I include bamtools in my code like this...
# include "api/BamAlignment.h"

2) Sally comes along and writes code to link against my library.  She doesn't know about the bamtools environment variables magic.

When she links against my headers, she's going to get build errors like,
"....api/BamAlignment.h: No such file or directory"

-- Solution? --

I took a peek at the code.  Many of the header files could refer to each other with relative paths.  Stripping the "api/" prefix from most of the includes would solve about 90% of the problem.  "shared/foobar.h" from api/blah.h might be slightly different.

In my very humble opinion, when I install a library, I shouldn't have to modify my environment or write special build code to find it.

And, although really a minor nit, I think it's cleaner to write...
# include "bamtools/api/BamAlignment.h"

than
# include "api/BamAlignment.h"

Hopefully this solution wouldn't disrupt your build style too much, and would make life easier for me and future developers.  If I'm missing some glaring detail, please feel free to call me a fool :D
